src/pymoca/backends/casadi/api.py

```diff
@@ -301,6 +301,13 @@ def load_model(model_folder: str, model_name: str, compiler_options: Dict[str, s
 
                 if ca.depends_on(ca.MX(delay_durations_raw[i]), all_symbols):
                     dur = delay_durations_raw[i]
+                    false_deps = ca.vertcat(*[
+                        var for var in ca.symvar(dur) if not ca.depends_on(dur, var)])
```

Comment:
That's an interesting way of doing it. This would hopefully also catch parameters and fixed inputs that aren't used in the durations. I do hope that `ca.depends_on` is accurate though, and stays accurate... I would've initially guessed that `depends_on` and e.g. `sparsity_jac()` would share code, but due to their different results that is maybe not the case.

Reply:
Yes, it was important to also catch the parameters and fixed inputs.
I agree, relying on depends_on() is a small gamble, but it is working well so far.